### ISSUE TITLE
Compile  Alt+D, Alt+F with an option "--disable-auto-clean"

### ISSUE THREAD

**Issue by m-chichikalov** [gepd/Deviot#200]
// Before report an issue, please make sure it hasn't been reported before. 
-> Look's like no one mention about that.

Hi there...
If use the short cuts in Deviot  (for instance, Compile  Alt+D, Alt+F) it always do it without --disable-auto-clean and compile all files and all libraries. That is really annoying... Can I fix it in some way?
Have a nice one.
Thanks in advance.

**Comment by m-chichikalov** [comment]
I simply added --disable-auto-clean in a list "cmd" in  compile.py . By the way I don't like this. What do you think would be a better approach?
![My changes](https://s8.hostingkartinok.com/uploads/images/2017/11/26248f77ce4a4d39ae714683a7d80955.png)

**Comment by gepd** [comment]
I could add an option in `Deviot > Compile/Upload Options` I think it could be the best way, what do you think?

**Comment by m-chichikalov** [comment]
Yes, it's Ok. However, maybe we should ask author of platformio to change "--disable-auto-clean" on to "--enable-auto-clean"? For me it looks more logical. I don't know...

**Comment by gepd** [comment]
you mean, change the behavior and make this option disabled by default?

**Comment by m-chichikalov** [comment]
Yes, but probably your decision better, it gives choice...

**Comment by gepd** [comment]
Yes, I'll add this option atp, I hope soon

**Comment by m-chichikalov** [comment]
If you don't have time for that I could take this upon myself. It works?

**Comment by m-chichikalov** [comment]
I installed  the deviot developer branch, and it looks like in this branch (or maybe because it uses PlatformIO 3.5.0rc2 ) both, compilation and uploading  go with disable cleaning.  
I double checked, my corrected files (compile.py and upload.py) were overwritten.

**Comment by gepd** [comment]
> If you don't have time for that I could take this upon myself. It works?

Yes, you can keep using what you showed in the image above.

> I installed the deviot developer branch, and it looks like in this branch (or maybe because it uses PlatformIO 3.5.0rc2 ) both, compilation and uploading go with disable cleaning.
> I double checked, my corrected files (compile.py and upload.py) were overwritten.

PlatformIO version is not depending on Deviot branch, you can select the PIO version in `Deviot Menu > Options > Use Development Pio Version`

I've not read the changelog of the PIO `3.5`, it may be a change in this new version
